feat: Support MariaDB as a database resource

Matt-B opened the pull request:
This PR adds MariaDB as a supported resource.

For testing purposes I've adapted the https://github.com/lydtechconsulting/springboot-postgres example project to use MariaDB, this can be found here: https://github.com/Matt-B/springboot-mariadb

rgolder1 commented:
Excellent!
Minor things:
1.  add logging to TestcontainersConfiguration to output the MariaDB config (down towards the bottom)
2. add a CHANGELOG entry (this will be a minor bump as new feature, so will become 3.1.0)
4. update the README 'Contents' to add link to the MariaDB section
5. update the README 'Supported Resources' to add MariaDB.
6. update the README 'Configuration Options' large table with the MariaDB config.
7. when you merge the PR, ensure the commit is squashed, and starts 'feat:', so that the minor bump auto-happens - (do not change the version in pom.xml yourself).

rgolder1 reviewed (commented):
Excellent!
Minor things:

1. add logging to TestcontainersConfiguration to output the MariaDB config (down towards the bottom)
2. add a CHANGELOG entry (this will be a minor bump as new feature, so will become 3.1.0)
3. update the README 'Contents' to add link to the MariaDB section
4. update the README 'Supported Resources' to add MariaDB.
5. update the README 'Configuration Options' large table with the MariaDB config.
6. when you merge the PR, ensure the commit is squashed, and starts 'feat:', so that the minor bump auto-happens - (do not change the version in pom.xml yourself).

Matt-B commented:
@rgolder1 Thanks for taking a look, I think I've addressed your points now as well as having done some testing by adapting one of the example projects to use MariaDB ([here](https://github.com/Matt-B/springboot-mariadb)). Can you take a look and see if you're happy to approve the changes?
